### ISSUE TITLE
Error when ConvertPDFToImages.execute is run

### ISSUE THREAD

**Issue by colord** [namuan/dr-doc-search#7]
I get the error below when running `dr-doc-search --train -i filename.pdf` on Windows 11.
I noticed that `convert_command` is set to this:
```python
convert_command = f"""convert -density 150 -trim -background white -alpha remove -quality 100 -sharpen 0x1.0 {input_file_page} -quality 100 {image_path}"""
```

I think convert might be an alias on Linux: [https://linux.die.net/man/1/convert](url)
but on Windows that alias is taken by an existing command: [https://www.wikiwand.com/en/Convert_(command)](url)
Is there a way to allow the use of `magick convert` for Windows machines?

```shell
Invalid Parameter - 150
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Python311\Scripts\dr-doc-search.exe\__main__.py", line 7, in <module>
  File "C:\Python311\Lib\site-packages\doc_search\app.py", line 48, in main
    run_workflow(context, training_workflow_steps())
  File "C:\Python311\Lib\site-packages\py_executable_checklist\workflow.py", line 36, in run_workflow
    __run_step(step, context)
  File "C:\Python311\Lib\site-packages\py_executable_checklist\workflow.py", line 29, in __run_step
    returned_context = step_instance.execute() or {}
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\doc_search\workflow\__init__.py", line 102, in execute
    run_command(convert_command)
  File "C:\Python311\Lib\site-packages\py_executable_checklist\workflow.py", line 9, in run_command
    return subprocess.check_output(command, shell=True).decode("utf-8")  # nosemgrep
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\subprocess.py", line 465, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\subprocess.py", line 569, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'convert -density 150 -trim -background white -alpha remove -quality 100 -sharpen 0x1.0 C:\Users\iqaco\OutputDir\dr-doc-search\wind-up-bird-chronicle\wind-up-bird-chronicle.pdf[1] -quality 100 C:\Users\iqaco\OutputDir\dr-doc-search\wind-up-bird-chronicle\images\output-1.png' returned non-zero exit status 4.```

**Comment by namuan** [comment]
Would it be like the following on Windows?

```
convert_command = f"""magick convert -density 150 -trim -background white -alpha remove -quality 100 -sharpen 0x1.0 {input_file_page} -quality 100 {image_path}"""
```

**Comment by namuan** [comment]
I may implement something to allow a windows user to provide the location of the `convert` binary through an environment variable. Something which seems to be recommended [here](https://imagemagick.org/Usage/windows/#convert_issue)

```
SET IMCONV="%PROGRAMFILES%\ImageMagick\Convert"
```

**Comment by namuan** [comment]
If you can check this branch which should pick up the path from `IMCONV` environment variable

https://github.com/namuan/dr-doc-search/tree/pr-7-windows-convert

**Comment by colord** [comment]
Thanks for working on this 👍

On my end, this command does work:
```python
convert_command = f"""magick convert -density 150 -trim -background white -alpha remove -quality 100 -sharpen 0x1.0 {input_file_page} -quality 100 {image_path}"""
```

However, the proposed solution with the environment variable is not working because there is not a `convert` binary in Windows (despite what we see in the ImageMagick docs). The `convert` binary is actually a command you run using the magick binary: `magick convert`.

FYI, I used this for my environment variable:
```shell
set IMCONV="%PROGRAMFILES%\ImageMagick-7.1.0-Q16-HDRI\magick"
```

And I think for Windows, the convert_command would need to have a trailing `convert` after the environment variable:
```python
convert_command = f"""{self.convert_command} convert -density 150 -trim -background white -alpha remove -quality 100 -sharpen 0x1.0 {input_file_page} -quality 100 {image_path}"""
```
See my proposed changes here: [https://github.com/namuan/dr-doc-search/pull/10](https://github.com/namuan/dr-doc-search/pull/10)

**Comment by namuan** [comment]
This is merged and available in `1.6.3`. Thanks for your help with resolving this issue.
